src/client/datascience/interactive-window/interactiveWindow.ts

```diff
@@ -629,7 +625,7 @@ export class InteractiveWindow implements IInteractiveWindowLoadable {
             'interactive.open',
             { preserveFocus: true },
             notebookDocument.uri,
-            this.notebookController?.id,
+            this.notebookControllerManager.getSelectedNotebookController(notebookDocument)?.id,
```

Comment:
I'm assuming these changes can be revereted now?
Would actually prefer to go back to the older API (if we can confirm the issues were with the Python extension)

Reply:
The older api was saving the controller. This queries for it at run time. Seems more robust this way to me.

Reply:
Aaah yes, users can switch the kernels, thats right.